src/routing/mod.rs

```diff
@@ -312,17 +311,24 @@ impl Routing {
         dst: DstLocation,
         content: Bytes,
     ) -> Result<()> {
+        if let DstLocation::EndUser(EndUser::Client { socket_id, .. }) = dst {
+            if let Some(socket_addr) = self.stage.state.lock().await.get_socket_addr(&socket_id) {
```

Comment:
if a socket_addr cannot be fetched for the socket_id, we will still try to send this UserMessage?
does it mean a `relay` ? or shall we return with an error?

Reply:
The assumption here is that if there is no `socket_addr` here, that means this message is for a client in _another section_, and thus it should be relayed. I was thinking that sending it with `UserMessage` would make it be relayed, is that so?
If we check that our prefix matches this client `public_key`, then a missing `socket_addr` would be some sort of error.